File: android/src/main/java/com/rnfs2/Utils/MediaStoreQuery.java
package com.rnfs2.Utils;

public class MediaStoreQuery {
  public String uri;
  public String fileName;
  public String relativePath;
  public String mediaType;

  public MediaStoreQuery(String contentURI, String contentFileName, String contentRelativePath, String contentMediaType) {
    uri = contentURI != null ? contentURI : "";
    fileName = contentFileName != null ? contentFileName : "";
    relativePath = contentRelativePath != null ? contentRelativePath : "";
    mediaType = contentMediaType;
  }
}

File: android/src/main/java/com/margelo/nitro/fs2/Utils/MediaStoreQuery.kt
package com.margelo.nitro.fs2.Utils

// Using a data class for simplicity and standard Kotlin features
data class MediaStoreQuery(
        val uri: String = "",
        val fileName: String = "",
        val relativePath: String = "",
        val mediaType: String?
) {
  // The primary constructor now handles the defaulting logic from the original Java constructor.
  // No secondary constructor is needed if direct data class instantiation is used.
}
